### PR TITLE
Get rid of native=true from ReanimatedModule

### PR DESCRIPTION
**Pull request by Latropos**
## Test plan
Let's notce that `NativeReanimatedModule.native` is equal to `!shouldBeUseWeb()`. We can simplify our code a little substituting it.

### CODE REVIEW

File: src/createAnimatedComponent/InlinePropManager.ts [hunk 1]
```diff
@@ -154,9 +154,9 @@ export class InlinePropManager {
       const shareableViewDescriptors =
         this._inlinePropsViewDescriptors.shareableViewDescriptors;
 
-      const maybeViewRef = NativeReanimatedModule.native
-        ? undefined
-        : ({ items: new Set([this]) } as ViewRefSet<any>); // see makeViewsRefSet
+      const maybeViewRef = shouldBeUseWeb()
```
Comment (tjzel) [inline src/createAnimatedComponent/InlinePropManager.ts:157]: No need to evaluate it here each time. Instead evaluate `IS_NATIVE` in top level of this file.

File: src/reanimated2/hook/useAnimatedStyle.ts [hunk 2]
```diff
@@ -453,7 +452,7 @@ For more, see the docs: \`https://docs.swmansion.com/react-native-reanimated/doc
 
   const { initial, remoteState, viewDescriptors } = initRef.current;
   const shareableViewDescriptors = viewDescriptors.shareableViewDescriptors;
-  const maybeViewRef = NativeReanimatedModule.native ? undefined : viewsRef;
+  const maybeViewRef = shouldBeUseWeb() ? viewsRef : undefined;
```
Comment (tjzel) [inline src/reanimated2/hook/useAnimatedStyle.ts:455]: Also evaluate it in the top level of the file.

File: src/reanimated2/WorkletEventHandler.ts [hunk 3]
```diff
@@ -39,7 +39,7 @@ export default class WorkletEventHandler<Event extends object> {
     this.viewTag = undefined;
     this.registrations = [];
 
-    if (!NativeReanimatedModule.native) {
+    if (shouldBeUseWeb()) {
```
Comment (tjzel) [inline src/reanimated2/WorkletEventHandler.ts:42]: top level

File: src/reanimated2/mutables.ts [hunk 4]
```diff
@@ -64,7 +65,7 @@ export function makeMutable<T>(
 ): SharedValue<T> {
   let value: T = initial;
   let syncDataHolder: ShareableSyncDataHolderRef<T> | undefined;
-  if (!oneWayReadsOnly && NativeReanimatedModule.native) {
+  if (!oneWayReadsOnly && !shouldBeUseWeb()) {
```
Comment (tjzel) [inline src/reanimated2/mutables.ts:68]: Definitely top level here.

**Review by tjzel — CHANGES_REQUESTED** [review]

**Review by tjzel — APPROVED** [review]
Looks good, but make sure to check if it works on Web and simulators

**Review by piaskowyk — CHANGES_REQUESTED** [review]
Everything look great, but please confirm that it works on Web and MacOS too

**Review by piaskowyk — APPROVED** [review]

**Pull request merged**